plugin/sampling/strategystore/adaptive/processor.go

```diff
@@ -95,147 +93,134 @@ type Processor struct {
 		calculateProbabilitiesLatency: metricsFactory.Timer(metrics.TimerOptions{Name: "calculate_probabilities"}),
 	}, nil
 }
 
-// GetSamplingStrategy implements Thrift endpoint for retrieving sampling strategy for a service.
-func (p *Processor) GetSamplingStrategy(_ context.Context, service string) (*api_v2.SamplingStrategyResponse, error) {
-	p.RLock()
-	defer p.RUnlock()
-	if strategy, ok := p.strategyResponses[service]; ok {
+// GetSamplingStrategy implements protobuf endpoint for retrieving sampling strategy for a service.
+func (ss *StrategyStore) GetSamplingStrategy(_ context.Context, service string) (*api_v2.SamplingStrategyResponse, error) {
+	ss.RLock()
+	defer ss.RUnlock()
+	if strategy, ok := ss.strategyResponses[service]; ok {
 		return strategy, nil
 	}
-	return p.generateDefaultSamplingStrategyResponse(), nil
+	return ss.generateDefaultSamplingStrategyResponse(), nil
 }
 
-// Start initializes and starts the sampling processor which regularly calculates sampling probabilities.
-func (p *Processor) Start() error {
+// Start initializes and starts the PostAggregator which regularly calculates sampling probabilities.
+func (p *PostAggregator) Start() error {
 	p.logger.Info("starting adaptive sampling processor")
 	if err := p.electionParticipant.Start(); err != nil {
 		return err
 	}
 	p.shutdown = make(chan struct{})
-	p.loadProbabilities()
-	p.generateStrategyResponses()
-	p.runBackground(p.runCalculationLoop)
-	p.runBackground(p.runUpdateProbabilitiesLoop)
 	return nil
 }
 
-func (p *Processor) runBackground(f func()) {
+func (p *PostAggregator) runBackground(f func()) {
 	p.bgFinished.Add(1)
 	go func() {
 		f()
 		p.bgFinished.Done()
 	}()
 }
 
-// Close stops the processor from calculating probabilities.
-func (p *Processor) Close() error {
+// Close stops the PostAggregator from calculating probabilities.
+func (p *PostAggregator) Close() error {
 	p.logger.Info("stopping adaptive sampling processor")
 	err := p.electionParticipant.Close()
-	if p.shutdown != nil {
-		close(p.shutdown)
-	}
 	p.bgFinished.Wait()
 	return err
 }
 
-func (p *Processor) loadProbabilities() {
+func (ss *StrategyStore) loadProbabilities() {
 	// TODO GetLatestProbabilities API can be changed to return the latest measured qps for initialization
-	probabilities, err := p.storage.GetLatestProbabilities()
+	probabilities, err := ss.storage.GetLatestProbabilities()
 	if err != nil {
-		p.logger.Warn("failed to initialize probabilities", zap.Error(err))
+		ss.logger.Warn("failed to initialize probabilities", zap.Error(err))
 		return
 	}
-	p.Lock()
-	defer p.Unlock()
-	p.probabilities = probabilities
+	ss.Lock()
+	defer ss.Unlock()
+	ss.probabilities = probabilities
 }
 
 // runUpdateProbabilitiesLoop is a loop that reads probabilities from storage.
 // The follower updates its local cache with the latest probabilities and serves them.
-func (p *Processor) runUpdateProbabilitiesLoop() {
+func (ss *StrategyStore) runUpdateProbabilitiesLoop() {
```

Comment:
did the linter force you to rename receiver p->ss? If not, I would just keep it as p to minimize code diffs